helm/prometheus-meta-operator/templates/deployment.yaml

```diff
@@ -33,6 +33,7 @@ spec:
           items:
           - key: secret.yml
             path: secret.yml
+      {{- if eq .Values.managementCluster.provider.flavor "vintage" }}
```

Comment:
How does that fix the issue ? Because if the secret does not exist, sure PMO will run but there will be no CA for etcd :)

Reply:
The secret is available in CAPI, but doesn't exists in Vintage

Reply:
Oh ok then maybe we need this if in the go code also where the etcd secret is read from the disk

Reply:
`pmo` will raise errors if secret not found:
https://github.com/giantswarm/prometheus-meta-operator/blob/master/service/controller/resource/etcd-certificates/source.go#L38

Reply:
if not, pmo will get the certs from Disk

Reply:
Yes but then why would PMO go through  sc.getSourceFromDisk()  on CAPI :D

Reply:
On CAPI there are no etcd certificates on disk according Pau

Reply:
What do you suggest?

Reply:
I think the code, is good. 
We just added a condition in helm, so we make sure that the etcd-certs binding are only for vintage.

Reply:
So my suggestion is that we add this check also in the go code because if  the etcd secret source namespace is not found on capi clusters, what will happen to the code? Will sc.getSourceFromDisk() error out or panic and crash PMO?

Reply:
If the secret is not found,  pmo will look at certificates on disk and if there are none, it will throw an error like then one we have today when certificates are seen as directories rather than files.

I'm adding a test on CAPI vs Vintage to throw the proper error.

Reply:
Sure I agree with you here ;)

Reply:
BTW I approved yesterday :)

Reply:
I added some code this morning with the check you suggested... Can you review it ?